### PR TITLE
Implement intrinsic `Trailz`

### LINKED ISSUE #2530: Implement intrinsic `Trailz`
```fortran
program main
print *, trailz(6)
end program
```

> GFortran

```console
% gfortran a.f90 && ./a.out
           1
```

Ref: https://gcc.gnu.org/onlinedocs/gcc-4.9.4/gfortran/TRAILZ.html

Implementation can follow following fortran function

```fortran
function trailz(n) result(result)
    integer :: n
    integer :: result
    result = 0
    do while (mod(n,2) == 0)
        n = n/2
        result = result + 1
    end do
end function
```

@Thirumalai-Shaktivel is it possible to call `mod` intrinsic in body of `trailz` intrinsic?

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #2530. Towards #2524.

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I am not sure why `CI / LFortran CI (3.10, windows-2019) (pull_request)` is failing.

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -2067,6 +2074,107 @@ namespace Mod {
...
+        ASR::ttype_t *type1 = ASRUtils::expr_type(x.m_args[0]);
+        ASRUtils::require_impl(is_integer(*type1),
+            "ASR Verify: Arguments to Trailz must be of integer type",
+            x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_Trailz(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*> &args) {
+        int64_t a = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        int64_t trailing_zeros = __builtin_ctz(a);
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/intrinsic_function_registry.h:2092]: Here is the reason for the failure
```
intrinsic_function_registry.h(2092): error C3861: '__builtin_ctz': identifier not found
```

Comment (certik) [reply src/libasr/pass/intrinsic_function_registry.h:2092]: Yes, you might need to implement this function by hand at compile time.

**Comment by Thirumalai-Shaktivel** [comment]
If we have Fortran example implementation, it seems very easy to implement (for developers) and verify (for reviewers) in IntrinsicFunction. I really like this design.

This PR looks excellent!
Let's implement eval_trailz, it seems very easy. Just convert the Fortran example code to C++ code.

**Comment by Pranavchiku** [comment]
Yes, I will do it either by tonight, or max by tomorrow 2pm :)

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -2067,6 +2074,107 @@ namespace Mod {
 
 } // namespace Mod
 
+namespace Trailz {
+
+     static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args == 1,
+            "ASR Verify: Call to Trailz must have exactly 2 arguments",
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/intrinsic_function_registry.h:2081]:
```suggestion
            "ASR Verify: Call to Trailz must have exactly 1 argument",
```

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Review by certik — APPROVED** [review]

**Comment by Pranavchiku** [comment]
Thanks for completing this PR! I had a test and then was travelling back home, so was unable to finish it.

**Pull request merged**